schema/types/isolator.go

```diff
@@ -40,6 +44,27 @@ func AddIsolatorName(n ACIdentifier, ns map[ACIdentifier]struct{}) {
 // and PodManifest schemas.
 type Isolators []Isolator
 
+// assertValid checks that every single isolator is valid and that
+// the whole set is well built
+func (isolators Isolators) assertValid() error {
+	typesMap := make(map[ACIdentifier]uint)
+	for _, i := range isolators {
+		if err := i.Value().AssertValid(); err != nil {
+			return err
+		}
+		if typesMap[i.Name]+1 > i.Value().MaxInstances() {
+			return fmt.Errorf(`isolators set contains too many instances of type %s"`, i.Name)
```

Comment:
Check out https://github.com/appc/spec/blob/master/schema/types/errors.go#L26 for an alternative way to go here